modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java

```diff
@@ -2873,8 +2868,22 @@ record = new TxRecord(tx.state(), tx.nearXidVersion(), tx.writeVersion(), nodes)
         }
     }
 
-    /** Checks if a record belongs to a persistent cache with WAL enabled. */
-    private boolean containsCacheWithEnabledWal(IgniteTxAdapter tx) {
+    /**
+     * Returns {@code true} if it is possible to skip logging of transaction record to WAL.
+     *
+     * @param tx Transaction.
+     * @return {@code true} if it is possible to skip logging of transaction record to WAL.
+     */
+    private boolean canSkipTxRecordLogging(IgniteTxAdapter tx) {
+        BaselineTopology baselineTop = cctx.kernalContext().state().clusterState().baselineTopology();
+
+        if (baselineTop == null || !baselineTop.consistentIds().contains(cctx.localNode().consistentId()))
+            return true;
+
+        // Point-in-time recovery is enabled, so we have to log all transaction markers.
+        if (cctx.snapshot().needTxReadLogging())
```

Comment:
Also regarding the method semantics: does it mean that, if `needTxReadLogging`, then tx logging cannot be skipped? I think this invertion is not needed, it makes the code complicated

Reply:
> Also regarding the method semantics: does it mean that, if needTxReadLogging, then tx logging cannot be skipped?

Yes, it does. 

> I think this invertion is not needed, it makes the code complicated

Ok. Fixed.